src/main/java/com/example/weatherwaveapi/service/WeatherRouterService.java

```diff
@@ -0,0 +1,22 @@
+package com.example.weatherwaveapi.service;
+
+import com.example.weatherwaveapi.serviceapienum.ServiceApiEnum;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class WeatherRouterService {
+    private static final String NOT_SUPPORTED_MESSAGE = "Not supported service";
+    private final OpenWeatherService openWeatherService;
+    private final YandexWeatherService yandexWeatherService;
+
+
+    public WeatherService getWeatherBySelectedService(ServiceApiEnum service) {
+        return switch (service) {
+            case OPEN_WEATHER_MAP -> openWeatherService;
+            case YANDEX -> yandexWeatherService;
+            default -> throw new IllegalArgumentException(NOT_SUPPORTED_MESSAGE);
```

Comment:
There is no point in issuing such a simple error message, it also complicates the code and can be misleading

Reply:
Ok, I did it